### PR TITLE
[reactstrap] Adding props export to all elements.

### PR DESCRIPTION
**Pull request by Sprogel**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: **N/A**
- [X] Increase the version number in the header if appropriate. **N/A**
- [X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. **N/A**

### CODE REVIEW

**Comment by FaithForHumans** [comment]
Looks like the bot didn't grab the contributors.
Tagging: @alihammad @mfal @danilobjr  @fabiopaiva

**Comment by RyanCavanaugh** [comment]
This PR has been open and unchanged 5 days without signoff or complaint. This will be merged by a maintainer soon if there are no objections.

File: types/reactstrap/lib/Alert.d.ts
Comment (DanielRosenwasser) [inline types/reactstrap/lib/Alert.d.ts]: This is problematic because it means that deep imports to the package will now break, even if these were actually default exports.

I think these should be reverted, and the `default`s should continue to be re-exported as before.

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

File: types/reactstrap/lib/Alert.d.ts
Comment (FaithForHumans) [reply types/reactstrap/lib/Alert.d.ts]: I didn't see anyone in importing reactstrap, but it's a good point. I'll go through and add all of the default exports back in. I'm going to wait until #21178 has been merged before I submit the commit.

**Comment by RyanCavanaugh** [comment]
@FaithForHumans Please address comments from the code reviewers.

**Comment by DanielRosenwasser** [comment]
Should be merged!

File: types/reactstrap/lib/Alert.d.ts
Comment (DanielRosenwasser) [reply types/reactstrap/lib/Alert.d.ts]: Should be merged!

File: types/reactstrap/lib/Badge.d.ts
Comment (DanielRosenwasser) [inline types/reactstrap/lib/Badge.d.ts]: Unfortunately, this isn't correct either; there are no named exports in `Badge.d.ts` and the like; they're all default exports.

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

File: types/reactstrap/lib/Badge.d.ts
Comment (FaithForHumans) [reply types/reactstrap/lib/Badge.d.ts]: I must have misunderstood what you were talking about then.

The default exports were left untouched in case someone else was referencing directly (it's still there, the diff isn't showing it). I added in named exports for the component in the file, as well as it's props.

What should it look like? Should the default export should be the only export in the file? If so, would the following be correct?
```
declare namespace Badge {
  interface Props {}
}
declare const Badge: React.StatelessComponent<Badge.Props>
export default Badge;
```

Comment (DanielRosenwasser) [reply types/reactstrap/lib/Badge.d.ts]: Yup, sorry I didn't get back to this sooner. That is more appropriate, but you can't have `Props` as a separate type as well:

```ts
export interface Props {

}

const Badge: React.StatelessComponent<Props>;
export default Badge;
```

Comment (FaithForHumans) [reply types/reactstrap/lib/Badge.d.ts]: I don't have an issue going through and updating everything, I just don't understand as to why it's incorrect / bad practice. Do you have an example of where the current definition could lead to issues?

What do you mean by: "you can't have `Props` as a separate type as well"? I don't understand the functional difference between your example and the current merge (unless you were saying what you couldn't do).

Comment (DanielRosenwasser) [reply types/reactstrap/lib/Badge.d.ts]: If you had the following

```ts
export const Badge: React.StatelessComponent<Props>;
export default Badge;
```

That means that you also have a named export which doesn't actually exist, so you could write something like

```ts
import BadgeDefault, { BadgeNamed } from "reactstrap/lib/Badge";

let a = <BadgeDefault /> // okay, should work
let b = <BadgeNamed /> // error at runtime
```

Comment (FaithForHumans) [reply types/reactstrap/lib/Badge.d.ts]: If the named export would cause run-time issues, then how come the tests are passing? When I re-exported everything in the index file, it's using the named export, not the default export.

I've also dropped the change into an existing project and it hasn't caused any issues. I can make a small test repo with these changes if that would help.

**Comment by sandersn** [comment]
@DanielRosenwasser do you mind taking another look at this? I didn't understand your previous comments well enough to take over the review.

**Comment by FaithForHumans** [comment]
Bump @DanielRosenwasser

File: types/reactstrap/lib/Badge.d.ts
Comment (DanielRosenwasser) [reply types/reactstrap/lib/Badge.d.ts]: The tests are passing for two reasons:

1. They don't test the runtime behavior.
2. There are no tests that import from anywhere other than the `index.d.ts` entry-point. For consumers who only consume from the entry-point, they'll never see a difference.

The `.d.ts` files themselves previously reflected [the way that the library was authored](https://github.com/reactstrap/reactstrap/blob/57d07e36b43b0b8197606fc884d3733fa25842d4/src/Breadcrumb.js#L36) - that each module within `lib` had a `default` export. For anyone who used to write

```ts
import Breadcrumb from "reactstrap/lib/Breadcrumb"
```

they'll simply get an error now. When they try to change it to

```ts
import { Breadcrumb } from "reactstrap/lib/Breadcrumb"
```

TypeScript won't report an error, but they'll get an error at runtime.

Comment (FaithForHumans) [reply types/reactstrap/lib/Badge.d.ts]: Ohh, okay. That makes sense now; I wasn't sure what you were getting at before. I'll have a fix shortly. Thanks for the clarification, I really appreciate it!

**Pull request merged**
